src/tpm2/Unmarshal.c

```diff
@@ -4437,6 +4509,121 @@ TPM2B_NV_PUBLIC_Unmarshal(TPM2B_NV_PUBLIC *target, BYTE **buffer, INT32 *size)
     return rc;
 }
 
+/* Table 226 - Definition of (UINT64) TPMA_NV_EXP Bits */
+
+TPM_RC
+TPMA_NV_EXP_Unmarshal(TPMA_NV_EXP *target, BYTE **buffer, INT32 *size)
+{
+    TPM_RC rc = TPM_RC_SUCCESS;
+
+    if (rc == TPM_RC_SUCCESS) {
+	rc = UINT64_Unmarshal(target, buffer, size);
+    }
+    if (rc == TPM_RC_SUCCESS) {
+	if (*target & TPMA_NV_EXP_reserved) {
+	    rc = TPM_RC_RESERVED_BITS;
+	}
+    }
+    return rc;
+}
+
+/* Table 229 - Definition of TPMS_NV_PUBLIC_EXP_ATTR Structure */
+
+TPM_RC
+TPMS_NV_PUBLIC_EXP_ATTR_Unmarshal(TPMS_NV_PUBLIC_EXP_ATTR *target, BYTE **buffer, INT32 *size)
+{
+    TPM_RC rc = TPM_RC_SUCCESS;
+
+    if (rc == TPM_RC_SUCCESS) {
+	rc = TPMI_RH_NV_EXP_INDEX_Unmarshal(&target->nvIndex, buffer, size);
+    }
+    if (rc == TPM_RC_SUCCESS) {
+	rc = TPMI_ALG_HASH_Unmarshal(&target->nameAlg, buffer, size, NO);
+    }
+    if (rc == TPM_RC_SUCCESS) {
+	rc = TPMA_NV_EXP_Unmarshal(&target->attributes, buffer, size);
+    }
+    if (rc == TPM_RC_SUCCESS) {
+	rc = TPM2B_DIGEST_Unmarshal(&target->authPolicy, buffer, size);
+    }
+    if (rc == TPM_RC_SUCCESS) {
+	rc = UINT16_Unmarshal(&target->dataSize, buffer, size);
+    }
+    if (rc == TPM_RC_SUCCESS) {
+	if (target->dataSize > MAX_NV_INDEX_SIZE) {
+	    rc = TPM_RC_SIZE;
+	}
+    }
+    return rc;
+}
+
+/* Table 230 - Definition of TPMU_NV_PUBLIC_2 Union */
+
+TPM_RC
+TPMU_NV_PUBLIC_2_Unmarshal( TPMU_NV_PUBLIC_2 *target, BYTE **buffer, INT32 *size, UINT8 selector)
+{
+    TPM_RC rc = TPM_RC_SUCCESS;
+
+    switch (selector) {
+      case TPM_HT_NV_INDEX:
+	rc = TPMS_NV_PUBLIC_Unmarshal(&target->nvIndex, buffer, size);
+	break;
+      case TPM_HT_EXTERNAL_NV:
+	rc = TPMS_NV_PUBLIC_EXP_ATTR_Unmarshal(&target->externalNV, buffer, size);
+	break;
+      case TPM_HT_PERMANENT_NV:
```

Comment:
Support for this will require an update to USER_NVRAM_Marshal and USER_NVRAM_Unmarshal